### PR TITLE
8.0 framework agreement sourcing stock route transit

### PR DESCRIPTION
**Pull request by gurneyalex**

### CODE REVIEW

**Comment by gurneyalex** [comment]
Needs https://github.com/OCA/stock-logistics-workflow/pull/76 to be fully effective

File: framework_agreement_sourcing_stock_route_transit/model/logistic_requisition_source.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                                line, po_pricelist, context=None):
+        """change the destination location of the generated purchase order
+
+        In case the PO delivers to a warehouse, use the incoming transit
+        location, if it is dropshipping, use the outgoing transit location.
+        """
+        IrModelData = self.pool['ir.model.data']
+
+        def ref(xmlid):
+            return IrModelData.xmlid_to_res_id(cr, uid, xmlid)
```
Comment (yvaucher) [inline framework_agreement_sourcing_stock_route_transit/model/logistic_requisition_source.py]: to replace by `self.env.ref()`

File: framework_agreement_sourcing_stock_route_transit/model/logistic_requisition_source.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+        _super = super(LogisticRequisitionSource, self)
+        res = _super._prepare_purchase_order(cr, uid,
+                                             line, po_pricelist,
+                                             context=context)
+        xmlid_dropship = 'stock_dropshipping.picking_type_dropship'
+        if res.get('picking_type_id') == ref(xmlid_dropship):
```
Comment (yvaucher) [inline framework_agreement_sourcing_stock_route_transit/model/logistic_requisition_source.py]: `self.env.ref().id`

File: framework_agreement_sourcing_stock_route_transit/model/logistic_requisition_source.py  [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, api
```
Comment (yvaucher) [inline framework_agreement_sourcing_stock_route_transit/model/logistic_requisition_source.py ]: api to remove or to use

File: framework_agreement_sourcing_stock_route_transit/model/logistic_requisition_source.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+import logging
+
+from openerp.osv import orm
+
+
+class LogisticRequisitionSource(orm.Model):
```
Comment (yvaucher) [inline framework_agreement_sourcing_stock_route_transit/model/logistic_requisition_source.py:26]: Any reason to make it in old api?

Comment (lepistone) [inline framework_agreement_sourcing_stock_route_transit/model/logistic_requisition_source.py:26]: So is the original in `framework_agreement_sourcing`. When we will migrate this method to the new api, we'll want to make this an instance method and get rid of the `line` parameters.

I suggest we keep that now, and migrate the two at a later time.

**Comment by lepistone** [comment]
Rebased to get rid of a ghost file ending with a space.

File: framework_agreement_sourcing_stock_route_transit/model/logistic_requisition_source.py  (hunk 3 above)
Comment (lepistone) [inline framework_agreement_sourcing_stock_route_transit/model/logistic_requisition_source.py ]: fixed

**Comment by lepistone** [comment]
I :+1: this, thanks. @yvaucher does it look good to you?

**Comment by yvaucher** [comment]
@lepistone Ok thanks for the fix

:+1:

**Comment by lepistone** [comment]
I'll merge anyway because tests were green before two trivial changes of mine (I removed an unused file, and changed the author).

**Pull request merged**